src/Adapter/MSTest.CoreAdapter/MSTestSettings.cs

```diff
@@ -338,6 +344,26 @@ private static MSTestSettings ToSettings(XmlReader reader)
                                 break;
                             }
 
+                        case "CLASSCLEANUPLIFECYCLE":
+                            {
+                                var value = reader.ReadInnerXml();
+                                if (Enum.TryParse(value, out ClassCleanupLifecycle lifecycle))
```

Comment:
Make this case insensitive, and check the the parsed value IsDefined, otherwise any value that is parsable to the underlying type (int) would be a valid enum value, e.g. "10".
```suggestion
                                if (Enum.TryParse(value, true, out ClassCleanupLifecycle lifecycle) && Enum.IsDefined(lifecycle))
```